### PR TITLE
Add support for N

### PR DESCRIPTION
**Pull request by Julien-R44**
The generation script works great man, it's awesome 🚀

Everything seems to work on my end correctly, I tested with a pnpm link, and everything is typed.

If you validate that I've done everything correctly, I'll add a couple of other plugins as well

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +64,7 @@
     "eslint-plugin-inclusive-language": "~2.2.0",
     "eslint-plugin-jsdoc": "~39.3.6",
     "eslint-plugin-mdx": "~2.0.2",
+    "eslint-plugin-n": "^15.3.0",
```
Comment (Shinigami92) [inline package.json:67]:
```suggestion
    "eslint-plugin-n": "~15.3.0",
```

Please re-run pnpm install after that to regenerate lock file

File: src/extends/index.d.ts [hunk 2]
```diff
@@ -21,6 +22,7 @@ export type KnownExtensions = LiteralUnion<
   | JsdocExtensions
   | MdxExtensions
   | NodeExtensions
+  | NExtensions
```
Comment (Shinigami92) [inline src/extends/index.d.ts:25]: Please run VSCode sort lines for this

File: src/rules/n/no-extraneous-import.d.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+export interface NoExtraneousImportOption {
+  allowModules?: string[];
+  convertPath?:
+    | {
+        /**
+         * @minItems 2
+         * @maxItems 2
+         *
+         * This interface was referenced by `undefined`'s JSON-Schema definition
+         * via the `patternProperty` "^.+$".
```
Comment (Shinigami92) [inline src/rules/n/no-extraneous-import.d.ts:15]: I personally go manually through the generated files and remove some garbage manually afterwards
Still not satisfied with my process, but this is coming from a JSONSchema generator sub dependency :confused:

**Review by Shinigami92 — CHANGES_REQUESTED** [review]
Looking good :+1: 
You definitely did the generate process right :slightly_smiling_face: 

Need to refresh my mind and check if you didn't missed anything outside the rules/extends folder.
e.g. settings or parser-options need to be manually added and are not covered by the generate process.

File: src/rules/n/no-extraneous-import.d.ts (hunk 3 above)
Comment (Julien-R44) [reply src/rules/n/no-extraneous-import.d.ts]: Dude, if that's the only thing we have to do, that's awesome. 99% of the boring work is done automatically. Then we can remove that thing automatically pretty easily too

Comment (Shinigami92) [reply src/rules/n/no-extraneous-import.d.ts]: > Dude, if that's the only thing we have to do, that's awesome. 99% of the boring work is done automatically. Then we can remove that thing automatically pretty easily too ( I guess ? )

Exactly :heart:

**Comment by Shinigami92** [comment]
Run `pnpm run lint && pnpm run format && pnpm run check` :slightly_smiling_face:

**Comment by Julien-R44** [comment]
Dude it's weird but I don't have this error in local 
image

Even a pnpm lint --fix does not change anything.

Will try to fix it manually

**Comment by Shinigami92** [comment]
:partying_face: 

I will check and merge it tomorrow, hope that's okay

**Comment by Julien-R44** [comment]
Well, that's super weird. This is the first time I've seen this. a lint error in CI, but not in local. Anyway !

**Comment by Julien-R44** [comment]
Awesome, thanks for the review dude!

**Comment by Shinigami92** [comment]
I have updated all dependencies and generated some updated rules for the existing ones.
Please update your PRs / rebase them.

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**